File: php/features.php
<?php
declare(strict_types=1);

// DnsLookup SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class DnsLookupFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new DnsLookupBaseFeature();
            case "test":
                return new DnsLookupTestFeature();
            default:
                return new DnsLookupBaseFeature();
        }
    }
}

File: py/features.py
# DnsLookup SDK feature factory

from feature.base_feature import DnsLookupBaseFeature
from feature.test_feature import DnsLookupTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DnsLookupBaseFeature(),
        "test": lambda: DnsLookupTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
